pkg/datastore/postgres/repositories/header_sync_log_repository.go

```diff
@@ -0,0 +1,146 @@
+// VulcanizeDB
+// Copyright © 2019 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package repositories
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+	"github.com/sirupsen/logrus"
+	"github.com/vulcanize/vulcanizedb/pkg/core"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+)
+
+const insertHeaderSyncLogQuery = `INSERT INTO header_sync_logs
+		(header_id, address, topics, data, block_number, block_hash, tx_index, tx_hash, log_index, raw)
+		VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) ON CONFLICT DO NOTHING`
+
+type HeaderSyncLogRepository struct {
+	db                *postgres.DB
+	addressRepository AddressRepository
+}
+
+func NewHeaderSyncLogRepository(db *postgres.DB) HeaderSyncLogRepository {
+	return HeaderSyncLogRepository{
+		db:                db,
+		addressRepository: AddressRepository{},
+	}
+}
+
+type headerSyncLog struct {
+	ID          int64
+	HeaderID    int64 `db:"header_id"`
+	Address     int64
+	Topics      pq.ByteaArray
+	Data        []byte
+	BlockNumber uint64 `db:"block_number"`
+	BlockHash   string `db:"block_hash"`
+	TxHash      string `db:"tx_hash"`
+	TxIndex     uint   `db:"tx_index"`
+	LogIndex    uint   `db:"log_index"`
+	Transformed bool
+	Raw         []byte
+}
+
+func (repository HeaderSyncLogRepository) GetUntransformedHeaderSyncLogs() ([]core.HeaderSyncLog, error) {
+	rows, queryErr := repository.db.Queryx(`SELECT * FROM public.header_sync_logs WHERE transformed = false`)
```

Comment:
since this method is checking for header_sync_logs `where transformed = false`, where are we setting transformed to true?

Reply:
That's a responsibility of the event transformer repository's `Create` function in this setup

Reply:
oh gotcha!

Reply:
@rmulhol Any reason not to move that responsibility up to the watcher in case of a non-error execution of the transformer on that log? :>

Reply:
Guess I just wanted to keep the field switching as close as possible to the place where the truth changed, but I'd be down to move it up if we prioritize not having plugins write back to tables in the `public` schema. Only concern would be that it maybe prevents us from achieving max async (i.e. rn the delegator could theoretically ignore transformer errors, just send logs and move on - that's not true if it's responsible for flipping fields based on errors in transformer execution)

Reply:
@rmulhol Makes sense. Still kinda feels like we could reduce transformer complexity further here though, if we assume they throw errors on failure (which they do)